File: packages/urdu-js/src/tokenization/ngrams.ts
export function ngrams(tokens: string[], n: number): string[][] {
  if (n <= 0 || tokens.length < n) return []
  const result: string[][] = []
  for (let i = 0; i <= tokens.length - n; i++) result.push(tokens.slice(i, i + n))
  return result
}

File: packages/urdu-dart/lib/src/tokenization/ngrams.dart
/// Generates word-level n-grams from a list of token strings.
List<List<String>> ngrams(List<String> tokens, int n) {
  if (tokens.length < n) return [];
  final result = <List<String>>[];
  for (var i = 0; i <= tokens.length - n; i++) {
    result.add(tokens.sublist(i, i + n));
  }
  return result;
}
